FilmFestivalLoader/IDFA/parse_idfa_html.py

```diff
@@ -384,291 +169,74 @@ def load_az_pages(festival_data):
+    if theme_html:
+        comment(f'Analysing {theme_str} theme page, encoding={url_file.encoding}')
+        ThemePartsPageParser(festival_data, theme_str).feed(theme_html)
+
+
+def get_films_by_theme(festival_data, theme_urls, theme_str):
+    """Read the films from each of the given pathway URLs."""
+    comment(f'Finding films per {theme_str} theme ({len(theme_urls)} {theme_str}s)')
+    for i, theme_url in enumerate(theme_urls):
+        theme_file = FILE_KEEPER.numbered_webdata_file(theme_str, i)
+        url_file = UrlFile(theme_url, theme_file, ERROR_COLLECTOR, DEBUG_RECORDER, byte_count=200)
+        comment_ = f'Downloading {theme_url} as to find the {theme_str} parts of the encountered films'
+        theme_html = url_file.get_text(always_download=ALWAYS_DOWNLOAD, comment_at_download=comment_)
+        if theme_html:
+            comment(f'Analysing {theme_str} page {i}, encoding={url_file.encoding}')
+            match theme_str:
+                case('sections'):
+                    FilmsFromSectionPageParser(festival_data, theme_url).feed(theme_html)
+                case(_):
+                    FilmsFromPathwayPageParser(festival_data, theme_url).feed(theme_html)
+
+
+def get_readable_theme_str(pathway_url):
+    pathway_str = pathway_url.split('/')[-2]  # .replace('_', ' ')
+    return pathway_str
+
+
+def get_film_from_theme_part_page(festival_data, film_title, film_url, theme_part_url, use_section_keeper=True):
+    """Store the sections of a film that was found in the given theme page."""
+    # Check if the film is already loaded.
+    try:
+        _ = festival_data.get_film_by_key(film_title, film_url)
+    except KeyError:
+        DEBUG_RECORDER.add('New film.')
+    except ValueError:
+        DEBUG_RECORDER.add('Known film not yet parsed.')
+    else:
+        comment(f'Existing film {film_title} not parsed.')
+        return
+
+    # Load the pathway film.
+    theme_str = get_readable_theme_str(theme_part_url)
+    comment(f'Parsing film "{film_title}" with theme "{theme_str}".')
+    # TODO in 2026: Stop using a "section keeper" to bookkeep film ID's.
+    if use_section_keeper:
+        film_id = SectionsKeeper.get_new_film_id(film_url)
+    else:
+        film_id = festival_data.new_film_id(festival_data.film_key(film_title, film_url)) + 1000
+    theme_film_file = FILE_KEEPER.film_webdata_file(film_id)
+    url_file = UrlFile(film_url, theme_film_file, ERROR_COLLECTOR, DEBUG_RECORDER, byte_count=200)
+    comment_at_download = f'Downloading "{theme_str}" film "{film_title}" data from {film_url}'
+    film_html = url_file.get_text(always_download=ALWAYS_DOWNLOAD, comment_at_download=comment_at_download)
+    if film_html:
+        message = (f'Analysing page of "{film_title}" from theme "{theme_str}", '
+                   f'parsed from "{theme_film_file}" encoding={url_file.encoding}')
+        print(message)
+        DEBUG_RECORDER.add(message)
```

Comment:
Please consider to add an application tool that takes a message as argument and writes that message to both the console and the debug file.

Reply:
Done.
Called it `broadcast()`